SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosStatusUpdateHandler.java

```diff
@@ -175,6 +176,10 @@ private void saveNewTaskStatusHolder(SingularityTaskId taskIdObj, SingularityTas
     return Optional.absent();
   }
 
+  private void relaunchTask(SingularityTask task) {
+    taskManager.savePendingTask(task.getTaskRequest().getPendingTask());
```

Comment:
Is this actually how to put something on the pending queue? Not convinced this is correct

Reply:
I'm iffy on having this here. This is partially because we let the status update handler write to pending tasks, and partially because it means we are reusing a pending task id. Generally we let the SingularityScheduler do all of the work of creating a pending task to keep responsibility for those types of operations separate. We actually removed bits from the status update handler a little while back so that it would avoid mutating the pending task queue.

Instead I'd suggest using `requestManager` to add to the pending request queue. This will let the scheduler do it's normal thing and rebuild a full new pending task with new unique ID from that pending request